internal/services/tesla_task_service.go

```diff
@@ -9,20 +9,21 @@ import (
 	"github.com/DIMO-Network/devices-api/internal/config"
 	"github.com/DIMO-Network/devices-api/models"
 	"github.com/DIMO-Network/shared"
+	"github.com/DIMO-Network/shared/sdtask"
 	"github.com/Shopify/sarama"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/segmentio/ksuid"
 )
 
 //go:generate mockgen -source tesla_task_service.go -destination mocks/tesla_task_service_mock.go
 
 type TeslaTaskService interface {
-	StartPoll(vehicle *TeslaVehicle, udai *models.UserDeviceAPIIntegration, apiVersion int, region string) error
+	StartPoll(udai *models.UserDeviceAPIIntegration, sd *models.SyntheticDevice) error
```

Comment:
A bit confusing "StartPoll" name it looks more like produce the message instead of the poll.

Reply:
It sends out the task description to be executed.

Reply:
Maybe just `StartTask` and provide the task's type parameter. Though, I'm not sure if we will add more tasks to TeslaTaskService in the future.